src/core/thread/mesh_forwarder.cpp

```diff
@@ -1345,6 +1342,13 @@ void MeshForwarder::HandleFragment(uint8_t *aFrame, uint8_t aFrameLength,
 
     // copy Fragment
     message->Write(message->GetOffset(), aFrameLength, aFrame);
+
+    if (fragmentHeader->GetDatagramOffset() == 0)
+    {
+        // Security Check (only if this is the first fragment)
+        VerifyOrExit(mNetif.GetIp6Filter().Accept(*message), error = kThreadError_Drop);
```

Comment:
Good point... I will change it.
Another thing I was wondering: We should have the full TCP header in the first fragment, is there any case this may not be true?